lib/element_filter.rb

```diff
@@ -20,35 +20,48 @@ class ElementFilter
 
   def initialize(path, matchers = nil)
     @path = path
-    @matchers = []
+    @matchers = {}
 
-    raise("Wrong type") if ![NilClass, String, Array].include?(matchers.class)
+    if ![NilClass, Hash].include?(matchers.class)
+      raise Machinery::Errors::InvalidFilter.new("Wrong filter type")
+    end
 
     add_matchers(matchers) if matchers
   end
 
-  def add_matchers(matchers)
-    @matchers += Array(matchers)
+  def add_matchers(additional)
+    additional.each do |op, matchers|
+      operator = FilterOperator.new(op)
+      @matchers[operator] ||= []
+      @matchers[operator] += Array(matchers)
+    end
   end
 
   def matches?(value)
-    @matchers.each do |matcher|
-      case value
-      when Machinery::Array
-        value_array = value.elements
+    @matchers.each do |operator, matchers|
+      matchers.each do |matcher|
+        values_equal = case value
+        when Machinery::Array
+          value_array = value.elements
 
-        return true if (value_array - Array(matcher)).empty? && (Array(matcher) - value_array).empty?
-      when String
-        if matcher.is_a?(Array)
-          exception = Machinery::Errors::ElementFilterTypeMismatch.new
-          exception.failed_matcher = "#{path}=#{matcher.join(",")}"
-          raise exception
-        end
+          (value_array - Array(matcher)).empty? && (Array(matcher) - value_array).empty?
+        when String
+          if matcher.is_a?(Array)
+            exception = Machinery::Errors::ElementFilterTypeMismatch.new
+            exception.failed_matcher = "#{path}#{FilterOperator.to_string(operator)}#{matcher.join(",")}"
```

Comment:
Line is too long. [105/100]<br>Prefer single-quoted strings inside interpolations.